File: Client/cinema-app/src/app/core/models/movie-session.ts
import { MovieSessionDate } from '@core/models/movie-session-date';
import { TicketPrice } from '@core/models/ticket-price';

export interface MovieSession {
  id?: number,
  movieId: number,
  hallId: number,
  showTime: string,
  ticketPrices: TicketPrice[],
  movieSessionDates: MovieSessionDate[]
}

File: Server/Cinema/Cinema.Domain/Entities/MovieSession.cs
namespace CinemaApplication.Domain.Entities
{
    public class MovieSession
    {
        public int Id { get; set; }
        public DateTime ShowDateTime { get; set; }
        public int MovieId { get; set; }
        public Movie Movie { get; set; }
        public int HallId { get; set; }
        public Hall Hall { get; set; }
        public ICollection<MovieSessionDate> MovieSessionDates { get; set; }
        public ICollection<TicketPrice> TicketPrices { get; set; }
    }
}